### DISCUSSION TITLE
Any way to display variables in the simulation window?

### DISCUSSION THREAD

**Discussion by user2038493** [omnetpp/omnetpp#984] [Q&A] [answered]
Hello everyone, is there any way I can show my var + its values in the Qtenv simulation window? 

We have on the left all the parameters, gates, etc, but I would like to know if there is a way to display a var and the value it has, so that I don't have to do std::cout or EV all the time, and look at the bottom window/console

Thank you

**Comment by rhornig** [comment, marked as answer]
Sure. You can use the `WATCH()` macro to put any of your variables there. [See the manual](https://doc.omnetpp.org/omnetpp/manual/#sec:sim-lib:watches-and-snapshots) for details.

You can also display variable values in the module/submodule display string so it is continuously displayed on the graphical module canvas. Finally, you can even draw text (or anything else) on the module background using the Canvas API (see canvas example).

**Reply by user2038493** [reply]
Thank you very much, you are always so helpful
